api/src/main/java/org/openmrs/module/patientqueueing/model/PatientQueue.java

```diff
@@ -0,0 +1,150 @@
+package org.openmrs.module.patientqueueing.model;
+
+
+import org.openmrs.Patient;
+import org.openmrs.Provider;
+import org.openmrs.Location;
+import org.openmrs.Encounter;
+import org.openmrs.BaseOpenmrsData;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.Id;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Column;
+import javax.persistence.ManyToOne;
+import javax.persistence.JoinColumn;
+import java.io.Serializable;
+
+@Entity(name = "patientqueueing.PatientQueue")
+@Table(name = "patient_queue")
+public class PatientQueue extends BaseOpenmrsData implements Serializable {
+
+    @Id
+    @GeneratedValue
+    @Column(name = "patient_queue_id")
+    private Integer patientQueueId;
+
+    @ManyToOne
+    @JoinColumn(name = "patient_id")
+    private Patient patient;
+
+    @ManyToOne
+    @JoinColumn(name = "provider_id")
+    private Provider provider;
+
+    @ManyToOne
+    @JoinColumn(name = "location_from")
+    private Location locationFrom;
+
+    @ManyToOne
+    @JoinColumn(name = "location_to")
+    private Location locationTo;
+
+    @ManyToOne
+    @JoinColumn(name = "encounter_id")
+    private Encounter encounter;
+
+    @Column(name = "status", length = 255)
+    private String status;
+
+    @Column(name = "queue_number", length = 255)
+    private String queueNumber;
+
+    @Column(name = "priority")
+    private Integer priority;
```

Comment:
What values will priority have?

Reply:
This is a number assigned to the patient basing on urgency towards a review or treatment. For example Should be able to weight patients based on urgency.

Reply:
Is this a real or hypothetical requirement?